src/test/java/htsjdk/samtools/util/IOUtilTest.java

```diff
@@ -520,4 +527,240 @@ public void testIsBlockCompressedOnJimfs(Path file, boolean checkExtension, bool
+        final Path regExpDir = Files.createTempDirectory("regExpDir");
+        regExpDir.toFile().deleteOnExit();
+        final List<String> listExpected = Arrays.asList(expected);
+        final List<File> expectedFiles = new ArrayList<File>();
+        for (String name : allNames) {
+            final Path file = regExpDir.resolve(name);
+            file.toFile().createNewFile();
+            if (listExpected.contains(name)) {
+                expectedFiles.add(file.toFile());
+            }
+            file.toFile().deleteOnExit();
+
+        }
+        final File[] result = IOUtil.getFilesMatchingRegexp(regExpDir.toFile(), regexp);
+
+        Assert.assertEqualsNoOrder(result, expectedFiles.toArray());
+
+    }
+
+
+    @Test()
+    public void testReadLines() throws IOException {
+        final Path file = Files.createTempFile("tmp", ".txt");
+        file.toFile().deleteOnExit();
+        final int seed = 12394738;
+        final Random rand = new Random(seed);
+
+        final int nLines = 5;
+        final String[] lines = new String[nLines];
+        try (final BufferedWriter writer = Files.newBufferedWriter(file)) {
+            for (int i = 0; i < nLines; i++) {
+                final String line = TEST_STRING + Integer.toString(rand.nextInt(100000000));
+                lines[i] = line;
+                writer.write(line);
+                writer.newLine();
+            }
+        }
+        final IterableOnceIterator<String> retLines = IOUtil.readLines(file.toFile());
+        final Stream<String> expectedLines = Stream.of(lines);
+        final Iterator<String> itExpected = expectedLines.iterator();
+        while (retLines.hasNext() && itExpected.hasNext()) {
+            Assert.assertEquals(retLines.next(), itExpected.next());
+        }
+        Assert.assertFalse(retLines.hasNext());
+        Assert.assertFalse(itExpected.hasNext());
+
+
+    }
+
+    @DataProvider
+    public static Object[][] fileSuffixTests() {
+        return new Object[][]{
+                {TEST_DATA_DIR.resolve("ipsum.txt"), ".txt"},
+                {TEST_DATA_DIR.resolve("ipsum.txt.bgz"), ".bgz"},
+                {TEST_DATA_DIR, null}
+        };
+    }
+
+    @Test(dataProvider = "fileSuffixTests")
+    public void testSuffixTest(final Path file, final String expected) {
+        String ret = IOUtil.fileSuffix(file.toFile());
+        Assert.assertEquals(ret, expected);
+
+    }
+
+    @Test
+    public void testCopyDirectoryTree() throws IOException {
+        final Path copyToDir = Files.createTempDirectory("copyToDir");
+        copyToDir.toFile().deleteOnExit();
+        IOUtil.copyDirectoryTree(TEST_VARIANT_DIR.toFile(), copyToDir.toFile());
+        final Stream<Path> fileStream = Files.walk(TEST_VARIANT_DIR);
+        final Stream<Path> copyFileStream = Files.walk(copyToDir);
+
+        final Iterator<Path> itFileStream = fileStream.iterator();

```

Comment:
inline stream and iterator

Reply:
you can simplify this by actually using the stream...
```java
final List<Path> collect = Files.walk(TEST_VARIANT_DIR).filter(f -> !f.equals(TEST_VARIANT_DIR)).collect(Collectors.toList());
```

and then compare with Assert.assertEquals()